blocks/api/registration.js

```diff
@@ -104,3 +111,34 @@ export function getBlockSettings( slug ) {
 export function getBlocks() {
 	return Object.values( blocks );
 }
+
+/**
+ * Validates the block settings.
+ *
+ * @param  {Object}  settings Block settings.
+ * @return {Boolean} Whether the block settings are valid.
+ */
+export function validateBlockSettings( settings ) {
+	if ( ! settings ) {
+		console.error(
+			'Block settings must specify a save() and edit() method for each block.'
+		);
+		return false;
+	}
+
+	if ( ! settings.save || 'function' !== typeof settings.save ) {
+		console.error(
+			'Block settings must specify a save() method for each block.'
+		);
+		return false;
+	}
+
+	if ( ! settings.edit || 'function' !== typeof settings.edit ) {
+		console.error(
+			'Block settings must specify a edit() method for each block.'
+		);
+		return false;
+	}
```

Comment:
I think this is fine for now, this sets our interface as requiring `save()` and `edit()`.

If there is buy-in to do more thorough validation, then I would probably redo this entire approach to become a bit more programmatic.  The more and more validation we add will make things more and more sprawly, so we will probably eventually want to introduce a schema.  A schema could double serve to help validate incoming attributes for an actual block as well.